### PR TITLE
feat(csharp): Add support for idempotency headers

### PR DESCRIPTION
**Pull request by Swimburger**
@coderabbitai summary

### CODE REVIEW

File: generators/csharp/sdk/src/options/IdempotentRequestOptionsGenerator.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                parameters: [],
+                return_: csharp.Type.reference(this.context.getHeadersClassReference()),
+                interfaceReference: this.context.getIdempotentRequestOptionsInterfaceClassReference(),
+                type: csharp.MethodType.INSTANCE,
+                body: csharp.codeblock((writer) => {
+                    writer.writeLine("return new Headers(new Dictionary<string, string>");
+                    writer.writeLine("{");
+                    writer.indent();
+                    for (const header of this.context.getIdempotencyHeaders()) {
+                        // TODO: find a way to check if a header is nullable, and add nullable check
```
Comment (Swimburger) [inline generators/csharp/sdk/src/options/IdempotentRequestOptionsGenerator.ts:43]: Is there an existing method to check whether something will be nullable?
- string
- object
- classes
- int?
- double?
but not
- int
- double
- etc.

I can't apply `?.` to everything as it won't compile for non-nullable types.

Comment (dsinghvi) [reply generators/csharp/sdk/src/options/IdempotentRequestOptionsGenerator.ts:43]: can you check if the type of the header is a container + optional ?

Comment (dsinghvi) [reply generators/csharp/sdk/src/options/IdempotentRequestOptionsGenerator.ts:43]: Don't see an existing method for it, but i could imagine it taking in a type reference and returning true/false

File: seed/csharp-sdk/idempotency-headers/src/SeedIdempotencyHeaders/Core/IIdempotentRequestOptions.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+namespace SeedIdempotencyHeaders.Core;
+
+internal interface IIdempotentRequestOptions : IRequestOptions
+{
+    public string IdempotencyKey { get; init; }
+    public int IdempotencyExpiration { get; init; }
+    internal Headers GetIdempotencyHeaders();
+}
```
Comment (dsinghvi) [inline seed/csharp-sdk/idempotency-headers/src/SeedIdempotencyHeaders/Core/IIdempotentRequestOptions.cs:8]: what's the benefit of having the interface here as opposed to `IdempotentRequstOptions` just inheriting from `IRequestOptions` ?

File: seed/csharp-sdk/oauth-client-credentials/src/SeedOauthClientCredentials/Core/Public/RequestOptions.cs [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 namespace SeedOauthClientCredentials;
 
-public partial class RequestOptions
+public partial class RequestOptions : IRequestOptions
```
[the remark this answers is not in the document]
Comment (Swimburger) [reply seed/csharp-sdk/oauth-client-credentials/src/SeedOauthClientCredentials/Core/Public/RequestOptions.cs:9]: Yes, and for future variations of request options. I'm using interfaces for the lack of multiple inheritance in C#.

File: seed/csharp-sdk/idempotency-headers/src/SeedIdempotencyHeaders/Core/IIdempotentRequestOptions.cs (hunk 2 above)
Comment (Swimburger) [reply seed/csharp-sdk/idempotency-headers/src/SeedIdempotencyHeaders/Core/IIdempotentRequestOptions.cs:8]: Another options variation may be introduced in the future that may inherit from `IRequestOptions` or from `IIdempotentRequestOptions`

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**